src/runtime/internal/dom.ts

```diff
@@ -168,19 +195,32 @@ export function claim_element(nodes, name, attributes, svg) {
 			for (let k = 0; k < remove.length; k++) {
 				node.removeAttribute(remove[k]);
 			}
-			return nodes.splice(i, 1)[0];
+
+			return node;
+		} else {
+			// Ignore hydration errors caused by empty text nodes
+			if (node.nodeType !== 3 || (node.data !== ' ' && node.data !== '')) {
+				console.error(`Hydration error: Expected node "${name}" but found`, node);
```

Comment:
we shouldn't call this an "error". even calling it a "warning" seems a bit strong since it's intended to work this way at least sometimes. I'd probably make it an `info`